jupyterhub/app.py

```diff
@@ -1351,6 +1351,9 @@ def cleanup(self):
 
     def write_config_file(self):
         """Write our default config to a .py config file"""
+        if not os.path.exists(self.config_file):
```

Comment:
It should check for the *directory*, not the file (`os.path.dirname(self.config_file)`). The file missing is fine (and likely).